cosmwasm/utils.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const { readFileSync } = require('fs');
+const { calculateFee, GasPrice } = require('@cosmjs/stargate');
+const { instantiate2Address } = require('@cosmjs/cosmwasm-stargate');
+const { getSaltFromKey } = require('../evm/utils');
+
+const pascalToSnake = (str) => {
+    return str.replace(/([A-Z])/g, (group) => `_${group.toLowerCase()}`).replace(/^_/, '');
+};
+
+const uploadContract = async (config, options, wallet, client) => {
+    const [account] = await wallet.getAccounts();
```

Comment:
```suggestion
    const [account] = await wallet.getAccounts();
    const { artifactPath, contractName, instantiate2, salt, contractName } = options;
```
I'd definitely recommend deconstructing. It's a lot easier to debug when some field is missing cuz you wouldn't have to go through the entire function to tell which are being used.